emake.mk

```diff
@@ -0,0 +1,101 @@
+# Minimum configuration
+#   PACKAGE_BASENAME: the basename that prefixes all Lisp files in this package
+
+ifndef PACKAGE_BASENAME
+$(error PACKAGE_BASENAME is not set)
+endif
+
+# User options
+#   EMACS_ARGS: [extra arguments for each invocation of emacs]
+#   (below; see README)
+PACKAGE_FILE            ?= $(PACKAGE_BASENAME).el
+PACKAGE_LISP            ?= $(filter-out %-pkg.el %-autoloads.el, $(wildcard $(PACKAGE_BASENAME)*.el))
+PACKAGE_TESTS           ?= $(wildcard test/*.el)
+PACKAGE_ARCHIVES        ?= gnu
+PACKAGE_TEST_ARCHIVES   ?= gnu
+
+# Then, make it easy to invoke Emacs with EMake loaded.
+EMAKE = PACKAGE_FILE="$(PACKAGE_FILE)" \
```

Comment:
what about adding a `@` before `PACKAGE_FILE` to avoid getting all the noise in the log?

Reply:
I can understand the desire to keep a clean log, but hiding information causes more problems than it solves in my experience. If we were to do this, I would want EMake to report its invocation pattern under the `emake:` prefix.

Actually I'm not sure I'd want to do even that – from the documentation of variable `command-line-args`, it doesn't look I can rely on it to hold the entire invocation.

Reply:
ok, I lost this battle :-).